Snakefile

```diff
@@ -146,9 +150,10 @@ def get_outputs():
     if 'atlas-experiment-summary' in config['tool'] or 'all' in config['tool'] and skip(config['accession'],'atlas_experiment_summary'):
         outputs.append(f"{config['accession']}-atlasExperimentSummary.Rdata")
     if 'baseline-heatmap' in config['tool'] or 'all-baseline' in config['tool'] and skip(config['accession'],'baseline-heatmap'):
-        outputs.extend(expand(f"{config['accession']}"+"-heatmap-{metric}.pdf", metric=get_metrics() ))
+        outputs.extend(expand(f"{config['accession']}"+"-heatmap-{metric}.pdf", metric=metrics ))
     if 'baseline-coexpression' in config['tool'] or 'all-baseline' in config['tool'] and skip(config['accession'],'baseline-coexpression'):   
-        outputs.extend(expand(f"{config['accession']}"+"-{metric}-coexpressions.tsv.gz", metric=get_metrics() )) 
+        outputs.extend(expand(f"{config['accession']}"+"-{metric}-coexpressions.tsv.gz", metric=metrics )) 
```

Comment:
This needs to be modified to check the number of columns in the expression files (TPM and FPKM should have the same number of columns), and only request the coexpression files if the column number is greater than 3. 

This is a lot simpler to do than trying to account for dynamic failures in the baseline_coexpression rule.

Reply:
FYI @pcm32

Reply:
Checkpoint is overly complex for situations where we know in advance the output files. It's very easy here to read the first line of the TSVs  for FPKM and TPM and decide based on that whether or not we can execute the coexpression steps. It would also be wasteful of resources to submit cluster jobs for this when we know in advance they will fail.